[BUG] cache_test.py failed w/ cache.serializer in spark 3.3.0

**Describe the bug**
w/ `--conf spark.sql.cache.serializer=com.nvidia.spark.ParquetCachedBatchSerializer` in spark330

Caused by: java.lang.IllegalArgumentException: For input string: "null"
```
[2022-02-06T19:19:19.124Z] =================================== FAILURES ===================================
[2022-02-06T19:19:19.124Z] [31m[1m_ test_passing_gpuExpr_as_Expr[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}] _[0m
[2022-02-06T19:19:19.124Z] 
[2022-02-06T19:19:19.124Z] enable_vectorized_conf = {'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'true'}
[2022-02-06T19:19:19.124Z] 
[2022-02-06T19:19:19.124Z]     @pytest.mark.parametrize('enable_vectorized_conf', enable_vectorized_confs, ids=idfn)
[2022-02-06T19:19:19.124Z]     @allow_non_gpu('CollectLimitExec')
[2022-02-06T19:19:19.124Z]     def test_passing_gpuExpr_as_Expr(enable_vectorized_conf):
[2022-02-06T19:19:19.124Z] >       assert_gpu_and_cpu_are_equal_collect(
[2022-02-06T19:19:19.124Z]             lambda spark : unary_op_df(spark, string_gen)
[2022-02-06T19:19:19.124Z]                 .select(f.col("a")).na.drop()
[2022-02-06T19:19:19.124Z]                 .groupBy(f.col("a"))
[2022-02-06T19:19:19.124Z]                 .agg(f.count(f.col("a")).alias("count_a"))
[2022-02-06T19:19:19.124Z]                 .orderBy(f.col("count_a").desc(), f.col("a"))
[2022-02-06T19:19:19.124Z]                 .cache()
[2022-02-06T19:19:19.124Z]                 .limit(50), enable_vectorized_conf)
[2022-02-06T19:19:19.124Z] 
[2022-02-06T19:19:19.124Z] [1m[31m../../src/main/python/cache_test.py[0m:34: 
[2022-02-06T19:19:19.124Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-02-06T19:19:19.124Z] [1m[31m../../src/main/python/asserts.py[0m:505: in assert_gpu_and_cpu_are_equal_collect
[2022-02-06T19:19:19.124Z]     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
[2022-02-06T19:19:19.124Z] [1m[31m../../src/main/python/asserts.py[0m:424: in _assert_gpu_and_cpu_are_equal
[2022-02-06T19:19:19.124Z]     run_on_cpu()
[2022-02-06T19:19:19.124Z] [1m[31m../../src/main/python/asserts.py[0m:410: in run_on_cpu
[2022-02-06T19:19:19.124Z]     from_cpu = with_cpu_session(bring_back, conf=conf)
[2022-02-06T19:19:19.124Z] [1m[31m../../src/main/python/spark_session.py[0m:86: in with_cpu_session
[2022-02-06T19:19:19.124Z]     return with_spark_session(func, conf=copy)
[2022-02-06T19:19:19.124Z] [1m[31m../../src/main/python/spark_session.py[0m:70: in with_spark_session
[2022-02-06T19:19:19.124Z]     ret = func(_spark)
[2022-02-06T19:19:19.124Z] [1m[31m../../src/main/python/asserts.py[0m:198: in <lambda>
[2022-02-06T19:19:19.124Z]     bring_back = lambda spark: limit_func(spark).collect()
[2022-02-06T19:19:19.124Z] [1m[31m/home/jenkins/agent/workspace/jenkins-rapids_it-3.3.x-SNAPSHOT-dev-github-37/jars/spark-3.3.0-SNAPSHOT-bin-hadoop3.2/python/lib/pyspark.zip/pyspark/sql/dataframe.py[0m:768: in collect
[2022-02-06T19:19:19.124Z]     sock_info = self._jdf.collectToPython()
[2022-02-06T19:19:19.124Z] [1m[31m/home/jenkins/agent/workspace/jenkins-rapids_it-3.3.x-SNAPSHOT-dev-github-37/jars/spark-3.3.0-SNAPSHOT-bin-hadoop3.2/python/lib/py4j-0.10.9.3-src.zip/py4j/java_gateway.py[0m:1321: in __call__
[2022-02-06T19:19:19.124Z]     return_value = get_return_value(
[2022-02-06T19:19:19.124Z] [1m[31m/home/jenkins/agent/workspace/jenkins-rapids_it-3.3.x-SNAPSHOT-dev-github-37/jars/spark-3.3.0-SNAPSHOT-bin-hadoop3.2/python/lib/pyspark.zip/pyspark/sql/utils.py[0m:190: in deco
[2022-02-06T19:19:19.124Z]     return f(*a, **kw)
[2022-02-06T19:19:19.124Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-02-06T19:19:19.124Z] 
[2022-02-06T19:19:19.124Z] answer = 'xro391'
[2022-02-06T19:19:19.124Z] gateway_client = <py4j.clientserver.JavaClient object at 0x7f66a2ebf130>
[2022-02-06T19:19:19.124Z] target_id = 'o390', name = 'collectToPython'
[2022-02-06T19:19:19.124Z] 
[2022-02-06T19:19:19.125Z]     def get_return_value(answer, gateway_client, target_id=None, name=None):
[2022-02-06T19:19:19.125Z]         """Converts an answer received from the Java gateway into a Python object.
[2022-02-06T19:19:19.125Z]     
[2022-02-06T19:19:19.125Z]         For example, string representation of integers are converted to Python
[2022-02-06T19:19:19.125Z]         integer, string representation of objects are converted to JavaObject
[2022-02-06T19:19:19.125Z]         instances, etc.
[2022-02-06T19:19:19.125Z]     
[2022-02-06T19:19:19.125Z]         :param answer: the string returned by the Java gateway
[2022-02-06T19:19:19.125Z]         :param gateway_client: the gateway client used to communicate with the Java
[2022-02-06T19:19:19.125Z]             Gateway. Only necessary if the answer is a reference (e.g., object,
[2022-02-06T19:19:19.125Z]             list, map)
[2022-02-06T19:19:19.125Z]         :param target_id: the name of the object from which the answer comes from
[2022-02-06T19:19:19.125Z]             (e.g., *object1* in `object1.hello()`). Optional.
[2022-02-06T19:19:19.125Z]         :param name: the name of the member from which the answer comes from
[2022-02-06T19:19:19.125Z]             (e.g., *hello* in `object1.hello()`). Optional.
[2022-02-06T19:19:19.125Z]         """
[2022-02-06T19:19:19.125Z]         if is_error(answer)[0]:
[2022-02-06T19:19:19.125Z]             if len(answer) > 1:
[2022-02-06T19:19:19.125Z]                 type = answer[1]
[2022-02-06T19:19:19.125Z]                 value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
[2022-02-06T19:19:19.125Z]                 if answer[1] == REFERENCE_TYPE:
[2022-02-06T19:19:19.125Z] >                   raise Py4JJavaError(
[2022-02-06T19:19:19.125Z]                         "An error occurred while calling {0}{1}{2}.\n".
[2022-02-06T19:19:19.125Z]                         format(target_id, ".", name), value)
[2022-02-06T19:19:19.125Z] [1m[31mE                   py4j.protocol.Py4JJavaError: An error occurred while calling o390.collectToPython.[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 4.0 failed 1 times, most recent failure: Lost task 0.0 in stage 4.0 (TID 26) (10.233.67.211 executor 0): java.lang.IllegalArgumentException: For input string: "null"[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at scala.collection.immutable.StringLike.parseBoolean(StringLike.scala:330)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at scala.collection.immutable.StringLike.toBoolean(StringLike.scala:289)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at scala.collection.immutable.StringLike.toBoolean$(StringLike.scala:289)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at scala.collection.immutable.StringOps.toBoolean(StringOps.scala:33)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(ParquetSchemaConverter.scala:63)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.$anonfun$convertCachedBatchToInternalRowIter$1(ParquetCachedBatchSerializer.scala:759)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer.withResource(ParquetCachedBatchSerializer.scala:262)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.convertCachedBatchToInternalRowIter(ParquetCachedBatchSerializer.scala:744)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.hasNext(ParquetCachedBatchSerializer.scala:724)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:363)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:890)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:890)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:365)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:329)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:136)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:507)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1475)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:510)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   [0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   Driver stacktrace:[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2618)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2567)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2566)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2566)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1182)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1182)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at scala.Option.foreach(Option.scala:407)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1182)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2806)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2748)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2737)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:952)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2228)[0m
[2022-02-06T19:19:19.125Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2249)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2268)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:490)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:443)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.CollectLimitExec.executeCollect(limit.scala:48)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3657)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3825)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3823)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3654)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at java.lang.reflect.Method.invoke(Method.java:498)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at py4j.Gateway.invoke(Gateway.java:282)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   Caused by: java.lang.IllegalArgumentException: For input string: "null"[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at scala.collection.immutable.StringLike.parseBoolean(StringLike.scala:330)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at scala.collection.immutable.StringLike.toBoolean(StringLike.scala:289)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at scala.collection.immutable.StringLike.toBoolean$(StringLike.scala:289)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at scala.collection.immutable.StringOps.toBoolean(StringOps.scala:33)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(ParquetSchemaConverter.scala:63)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.$anonfun$convertCachedBatchToInternalRowIter$1(ParquetCachedBatchSerializer.scala:759)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer.withResource(ParquetCachedBatchSerializer.scala:262)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.convertCachedBatchToInternalRowIter(ParquetCachedBatchSerializer.scala:744)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.hasNext(ParquetCachedBatchSerializer.scala:724)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:363)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:890)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:890)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:365)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:329)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:136)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:507)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1475)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:510)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2022-02-06T19:19:19.126Z] [1m[31mE                   	... 1 more[0m
[2022-02-06T19:19:19.126Z] 
[2022-02-06T19:19:19.126Z] [1m[31m/home/jenkins/agent/workspace/jenkins-rapids_it-3.3.x-SNAPSHOT-dev-github-37/jars/spark-3.3.0-SNAPSHOT-bin-hadoop3.2/python/lib/py4j-0.10.9.3-src.zip/py4j/protocol.py[0m:326: Py4JJavaError
[2022-02-06T19:19:19.126Z] ----------------------------- Captured stdout call -----------------------------
[2022-02-06T19:19:19.126Z] ### CPU RUN ###
[2022-02-06T19:19:19.126Z] [31m[1m_ test_passing_gpuExpr_as_Expr[{'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}] _[0m
[2022-02-06T19:19:19.126Z] 
[2022-02-06T19:19:19.126Z] enable_vectorized_conf = {'spark.sql.inMemoryColumnarStorage.enableVectorizedReader': 'false'}
[2022-02-06T19:19:19.126Z] 
[2022-02-06T19:19:19.126Z]     @pytest.mark.parametrize('enable_vectorized_conf', enable_vectorized_confs, ids=idfn)
[2022-02-06T19:19:19.126Z]     @allow_non_gpu('CollectLimitExec')
[2022-02-06T19:19:19.126Z]     def test_passing_gpuExpr_as_Expr(enable_vectorized_conf):
[2022-02-06T19:19:19.126Z] >       assert_gpu_and_cpu_are_equal_collect(
[2022-02-06T19:19:19.126Z]             lambda spark : unary_op_df(spark, string_gen)
[2022-02-06T19:19:19.126Z]                 .select(f.col("a")).na.drop()
[2022-02-06T19:19:19.126Z]                 .groupBy(f.col("a"))
[2022-02-06T19:19:19.126Z]                 .agg(f.count(f.col("a")).alias("count_a"))
[2022-02-06T19:19:19.126Z]                 .orderBy(f.col("count_a").desc(), f.col("a"))
[2022-02-06T19:19:19.126Z]                 .cache()
[2022-02-06T19:19:19.126Z]                 .limit(50), enable_vectorized_conf)
[2022-02-06T19:19:19.126Z] 
[2022-02-06T19:19:19.126Z] [1m[31m../../src/main/python/cache_test.py[0m:34: 
[2022-02-06T19:19:19.127Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-02-06T19:19:19.127Z] [1m[31m../../src/main/python/asserts.py[0m:505: in assert_gpu_and_cpu_are_equal_collect
[2022-02-06T19:19:19.127Z]     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
[2022-02-06T19:19:19.127Z] [1m[31m../../src/main/python/asserts.py[0m:424: in _assert_gpu_and_cpu_are_equal
[2022-02-06T19:19:19.127Z]     run_on_cpu()
[2022-02-06T19:19:19.127Z] [1m[31m../../src/main/python/asserts.py[0m:410: in run_on_cpu
[2022-02-06T19:19:19.127Z]     from_cpu = with_cpu_session(bring_back, conf=conf)
[2022-02-06T19:19:19.127Z] [1m[31m../../src/main/python/spark_session.py[0m:86: in with_cpu_session
[2022-02-06T19:19:19.127Z]     return with_spark_session(func, conf=copy)
[2022-02-06T19:19:19.127Z] [1m[31m../../src/main/python/spark_session.py[0m:70: in with_spark_session
[2022-02-06T19:19:19.127Z]     ret = func(_spark)
[2022-02-06T19:19:19.127Z] [1m[31m../../src/main/python/asserts.py[0m:198: in <lambda>
[2022-02-06T19:19:19.127Z]     bring_back = lambda spark: limit_func(spark).collect()
[2022-02-06T19:19:19.127Z] [1m[31m/home/jenkins/agent/workspace/jenkins-rapids_it-3.3.x-SNAPSHOT-dev-github-37/jars/spark-3.3.0-SNAPSHOT-bin-hadoop3.2/python/lib/pyspark.zip/pyspark/sql/dataframe.py[0m:768: in collect
[2022-02-06T19:19:19.127Z]     sock_info = self._jdf.collectToPython()
[2022-02-06T19:19:19.127Z] [1m[31m/home/jenkins/agent/workspace/jenkins-rapids_it-3.3.x-SNAPSHOT-dev-github-37/jars/spark-3.3.0-SNAPSHOT-bin-hadoop3.2/python/lib/py4j-0.10.9.3-src.zip/py4j/java_gateway.py[0m:1321: in __call__
[2022-02-06T19:19:19.127Z]     return_value = get_return_value(
[2022-02-06T19:19:19.127Z] [1m[31m/home/jenkins/agent/workspace/jenkins-rapids_it-3.3.x-SNAPSHOT-dev-github-37/jars/spark-3.3.0-SNAPSHOT-bin-hadoop3.2/python/lib/pyspark.zip/pyspark/sql/utils.py[0m:190: in deco
[2022-02-06T19:19:19.127Z]     return f(*a, **kw)
[2022-02-06T19:19:19.127Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-02-06T19:19:19.127Z] 
[2022-02-06T19:19:19.127Z] answer = 'xro488'
[2022-02-06T19:19:19.127Z] gateway_client = <py4j.clientserver.JavaClient object at 0x7f66a2ebf130>
[2022-02-06T19:19:19.127Z] target_id = 'o487', name = 'collectToPython'
[2022-02-06T19:19:19.127Z] 
[2022-02-06T19:19:19.127Z]     def get_return_value(answer, gateway_client, target_id=None, name=None):
[2022-02-06T19:19:19.127Z]         """Converts an answer received from the Java gateway into a Python object.
[2022-02-06T19:19:19.127Z]     
[2022-02-06T19:19:19.127Z]         For example, string representation of integers are converted to Python
[2022-02-06T19:19:19.127Z]         integer, string representation of objects are converted to JavaObject
[2022-02-06T19:19:19.127Z]         instances, etc.
[2022-02-06T19:19:19.127Z]     
[2022-02-06T19:19:19.127Z]         :param answer: the string returned by the Java gateway
[2022-02-06T19:19:19.127Z]         :param gateway_client: the gateway client used to communicate with the Java
[2022-02-06T19:19:19.127Z]             Gateway. Only necessary if the answer is a reference (e.g., object,
[2022-02-06T19:19:19.127Z]             list, map)
[2022-02-06T19:19:19.127Z]         :param target_id: the name of the object from which the answer comes from
[2022-02-06T19:19:19.127Z]             (e.g., *object1* in `object1.hello()`). Optional.
[2022-02-06T19:19:19.127Z]         :param name: the name of the member from which the answer comes from
[2022-02-06T19:19:19.127Z]             (e.g., *hello* in `object1.hello()`). Optional.
[2022-02-06T19:19:19.127Z]         """
[2022-02-06T19:19:19.127Z]         if is_error(answer)[0]:
[2022-02-06T19:19:19.127Z]             if len(answer) > 1:
[2022-02-06T19:19:19.127Z]                 type = answer[1]
[2022-02-06T19:19:19.127Z]                 value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
[2022-02-06T19:19:19.127Z]                 if answer[1] == REFERENCE_TYPE:
[2022-02-06T19:19:19.127Z] >                   raise Py4JJavaError(
[2022-02-06T19:19:19.127Z]                         "An error occurred while calling {0}{1}{2}.\n".
[2022-02-06T19:19:19.127Z]                         format(target_id, ".", name), value)
[2022-02-06T19:19:19.127Z] [1m[31mE                   py4j.protocol.Py4JJavaError: An error occurred while calling o487.collectToPython.[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 9.0 failed 1 times, most recent failure: Lost task 0.0 in stage 9.0 (TID 53) (10.233.67.211 executor 0): java.lang.IllegalArgumentException: For input string: "null"[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at scala.collection.immutable.StringLike.parseBoolean(StringLike.scala:330)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at scala.collection.immutable.StringLike.toBoolean(StringLike.scala:289)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at scala.collection.immutable.StringLike.toBoolean$(StringLike.scala:289)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at scala.collection.immutable.StringOps.toBoolean(StringOps.scala:33)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(ParquetSchemaConverter.scala:63)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.$anonfun$convertCachedBatchToInternalRowIter$1(ParquetCachedBatchSerializer.scala:759)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer.withResource(ParquetCachedBatchSerializer.scala:262)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.convertCachedBatchToInternalRowIter(ParquetCachedBatchSerializer.scala:744)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.hasNext(ParquetCachedBatchSerializer.scala:724)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:363)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:890)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:890)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:365)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:329)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:136)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:507)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1475)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:510)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   [0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   Driver stacktrace:[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2618)[0m
[2022-02-06T19:19:19.127Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2567)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2566)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2566)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1182)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1182)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at scala.Option.foreach(Option.scala:407)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1182)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2806)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2748)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2737)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:952)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2228)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2249)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2268)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:490)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.executeTake(SparkPlan.scala:443)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.sql.execution.CollectLimitExec.executeCollect(limit.scala:48)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3657)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3825)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:109)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:169)[0m
[2022-02-06T19:19:19.128Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:95)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:779)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3823)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3654)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at java.lang.reflect.Method.invoke(Method.java:498)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at py4j.Gateway.invoke(Gateway.java:282)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at java.lang.Thread.run(Thread.java:748)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   Caused by: java.lang.IllegalArgumentException: For input string: "null"[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at scala.collection.immutable.StringLike.parseBoolean(StringLike.scala:330)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at scala.collection.immutable.StringLike.toBoolean(StringLike.scala:289)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at scala.collection.immutable.StringLike.toBoolean$(StringLike.scala:289)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at scala.collection.immutable.StringOps.toBoolean(StringOps.scala:33)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.sql.execution.datasources.parquet.ParquetToSparkSchemaConverter.<init>(ParquetSchemaConverter.scala:63)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.$anonfun$convertCachedBatchToInternalRowIter$1(ParquetCachedBatchSerializer.scala:759)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer.withResource(ParquetCachedBatchSerializer.scala:262)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.convertCachedBatchToInternalRowIter(ParquetCachedBatchSerializer.scala:744)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.ParquetCachedBatchSerializer$CachedBatchIteratorConsumer$$anon$3.hasNext(ParquetCachedBatchSerializer.scala:724)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:363)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:890)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:890)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:365)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:329)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.scheduler.Task.run(Task.scala:136)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:507)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1475)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:510)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)[0m
[2022-02-06T19:19:19.129Z] [1m[31mE                   	... 1 more[0m
```

--- a/sql-plugin/src/main/311+-all/scala/com/nvidia/spark/rapids/shims/v2/ParquetCachedBatchSerializer.scala
+++ b/sql-plugin/src/main/311+-all/scala/com/nvidia/spark/rapids/shims/v2/ParquetCachedBatchSerializer.scala
@@ -498,7 +498,6 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
                   case _ => false
                 },
                 (dataType, cv) => {
-                  //TODO: why do we have to copy to a vector
                   dataType match {
                     case d: DecimalType =>
                       withResource(cv.bitCastTo(DecimalUtil.createCudfDecimal(d))) {
@@ -1419,6 +1418,7 @@ protected class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer wi
 
     hadoopConf.setBoolean(SQLConf.PARQUET_BINARY_AS_STRING.key, false)
     hadoopConf.setBoolean(SQLConf.PARQUET_INT96_AS_TIMESTAMP.key, false)
+    hadoopConf.setBoolean(SQLConf.CASE_SENSITIVE.key, false)
 
     hadoopConf.set(ShimLoader.getSparkShims.parquetRebaseWriteKey,
       LegacyBehaviorPolicy.CORRECTED.toString)

